ngraph/test/pre_post_process.cpp

```diff
@@ -0,0 +1,137 @@
+}
+
+TEST(pre_post_process, element_type_and_scale) {
+    auto f = create_simple_function(element::i8, ngraph::Shape{1, 3, 4, 5});
+    ASSERT_EQ(f->get_output_element_type(0), element::i8);
+    ngraph::pass::VisualizeTree("before2.png").run_on_function(f);  // Visualize the nGraph function to an image
+    f = PrePostProcessor(f)
+            .in()
+            .tensor()
+            .set_element_type(element::f32)
+            .preprocess()
+            .scale(2.0f)
+            .scale(4.0f)
+            .network()
+            .set_layout(PartialLayout("NCHW"))
+            .build();
+    f->validate_nodes_and_infer_types();
+
+    ngraph::pass::VisualizeTree("after2.png").run_on_function(f);  // Visualize the nGraph function to an image
+    ASSERT_NE(f, nullptr);
+    ASSERT_EQ(f->get_output_element_type(0), element::i8);
+}
+
+TEST(pre_post_process, scale_vector_network_layout) {
+    auto f = create_simple_function(element::f32, ngraph::PartialShape{Dimension::dynamic(), 3, 4, 5});
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+    ngraph::pass::VisualizeTree("before2.png").run_on_function(f);  // Visualize the nGraph function to an image
+    f = PrePostProcessor(f)
+            .in()
+            .preprocess()
+            .scale({0.1f, 0.2f, 0.3f})
+            .network()
+            .set_layout(PartialLayout("NCHW"))
+            .build();
+    f->validate_nodes_and_infer_types();
+
+    ngraph::pass::VisualizeTree("after3.png").run_on_function(f);  // Visualize the nGraph function to an image
+    ASSERT_NE(f, nullptr);
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+}
+
+TEST(pre_post_process, scale_vector_tensor_layout) {
+    auto f = create_simple_function(element::f32, ngraph::PartialShape{Dimension::dynamic(), 4, 4, 3});
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+    ngraph::pass::VisualizeTree("before2.png").run_on_function(f);  // Visualize the nGraph function to an image
+    f = PrePostProcessor(f)
+            .in()
+            .tensor()
+            .set_layout(PartialLayout("NHWC"))
+            .preprocess()
+            .scale({0.1f, 0.2f, 0.3f})
+            .build();
+    f->validate_nodes_and_infer_types();
+
+    ngraph::pass::VisualizeTree("after4.png").run_on_function(f);  // Visualize the nGraph function to an image
+    ASSERT_NE(f, nullptr);
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+}
+
+TEST(pre_post_process, scale_vector_no_channels_layout) {
+    auto f = create_simple_function(element::f32, ngraph::PartialShape{Dimension::dynamic(), 4, 4, 3});
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+    ASSERT_ANY_THROW(f = PrePostProcessor(f)
+                             .in()
+                             .tensor()
+                             .set_layout(PartialLayout("NHW?"))
+                             .preprocess()
+                             .scale({0.1f, 0.2f, 0.3f})
+                             .build();
```

Comment:
can we add test that w/o layout scale throws exception?

Reply:
yes, will do